dissect/target/plugins/os/unix/modules.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from typing import Iterator, Optional
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export, internal
+from dissect.target.target import Target
+
+ModuleRecord = TargetRecordDescriptor(
+    "linux/module",
+    [
+        ("string", "name"),
+        ("varint", "size"),
+        ("varint", "refcount"),
+        ("stringlist", "modules_referred"),
+        ("uri", "source"),
+    ],
+)
+
+
+@dataclass
+class Module:
+    path: Optional[str] = None
+    name: Optional[str] = None
+    size: Optional[int] = None
+    refcnt: Optional[int] = None
+    modules_referred: Optional[list[str]] = None
```

Comment:
```suggestion
    path: str
    name: str
    size: int
    refcnt: int
    modules_referred: list[str]
```
As far as I can tell, none of these fields are actually optional and you always populate all of them.